packages/roosterjs-editor-dom/lib/list/VList.ts

```diff
@@ -207,8 +208,18 @@ export default class VList {
      * @param type Type of this list item, can be ListType.None
      */
     appendItem(node: Node, type: ListType) {
-        node = getTagOfNode(node) == 'LI' ? node : wrap(node, 'li');
-        this.items.push(type == ListType.None ? new VListItem(node) : new VListItem(node, type));
+        let newListNode = node;
+        const nodeTag = getTagOfNode(node);
+
+        if (nodeTag != 'LI' && nodeTag != 'TABLE') {
+            newListNode = changeElementTag(<HTMLElement>node, 'LI');
+        } else if (nodeTag == 'TABLE') {
```

Comment:
nit:
you can check in a better order:
if (table) { .. }
else if (not li) {
...}